### PR TITLE
Inconsistency nearest waypoint

### PR DESCRIPTION
**Pull request by iman512003**
This is to address #137 . Longitudinal_error is calculated between the position of the vehicle and the first waypoint,
error_5

### CODE REVIEW

File: smarts/core/waypoints.py [hunk 1]
```diff
@@ -300,6 +305,31 @@ def _equally_spaced_path(self, path):
             ref_waypoints_coordinates["ref_wp_headings"]
         )
 
+        first_wp_heading = ref_waypoints_coordinates["ref_wp_headings"][0]
```
Comment (JVillella) [inline smarts/core/waypoints.py:308]: Can we remove these extra newlines?

**Review by JVillella — APPROVED** [review]

File: smarts/core/waypoints.py [hunk 2]
```diff
@@ -300,6 +305,31 @@ def _equally_spaced_path(self, path):
...
+
+        wp_position = np.array([*path[0].wp.pos, 0])
+
+        vehicle_position = np.array([point[0], point[1], 0])
+
+        heading_vector = np.array(
```
Comment (davidrusu) [inline smarts/core/waypoints.py:314]: use `radians_to_vec` from `smarts.core.util` instead.

File: smarts/core/waypoints.py [hunk 3]
```diff
@@ -300,6 +305,31 @@ def _equally_spaced_path(self, path):
...
+            ]
+        )
+
+        projected_distant_wp_vehicle = np.inner(
+            (vehicle_position - wp_position), heading_vector
+        )
+
+        ref_waypoints_coordinates["ref_wp_positions_x"][0] = wp_position[
+            0
+        ] + projected_distant_wp_vehicle * math.cos(math.pi / 2 + first_wp_heading)
```
Comment (davidrusu) [inline smarts/core/waypoints.py:328]: Use the heading vector from above:
`wp_pos[0] + proj_dist * heading_vector[0]`

File: smarts/core/waypoints.py [hunk 4]
```diff
@@ -300,6 +305,31 @@ def _equally_spaced_path(self, path):
...
+
+        ref_waypoints_coordinates["ref_wp_positions_y"][0] = wp_position[
+            1
+        ] + projected_distant_wp_vehicle * math.sin(math.pi / 2 + first_wp_heading)
```
Comment (davidrusu) [inline smarts/core/waypoints.py:332]: same here, `heading_vector[1]`

**Review by davidrusu — CHANGES_REQUESTED** [review]
Some nits, graphs look great

File: smarts/core/waypoints.py (hunk 1 above)
Comment (iman512003) [reply smarts/core/waypoints.py:308]: Thanks, yes, of course.

File: smarts/core/waypoints.py (hunk 2 above)
Comment (iman512003) [reply smarts/core/waypoints.py:314]: Thanks, sure.

File: smarts/core/waypoints.py (hunk 3 above)
Comment (iman512003) [reply smarts/core/waypoints.py:328]: Thanks, sure.

File: smarts/core/waypoints.py (hunk 4 above)
Comment (iman512003) [reply smarts/core/waypoints.py:332]: Thanks, sure.

**Comment by iman512003** [comment]
> Some nits, graphs look great

Thanks you so much.

**Review by Gamenot — APPROVED** [review]
LGTM

**Pull request merged**